modules/fledge.js

```diff
@@ -0,0 +1,75 @@
+import { Fledge } from '@magnite/fledge.polyfill/dist/api/cjs/index.js';
+import { auctionManager } from '../src/auctionManager.js';
+import { config } from '../src/config.js';
+import CONSTANTS from '../src/constants.json';
+import { getGlobal } from '../src/prebidGlobal.js';
+import * as utils from '../src/utils.js';
+
+/**
+ * @summary This Module is intended to provide users with the ability to run a Fledge-based auction against a PreBid winning bid
+ */
+const MODULE_NAME = 'Fledge';
+
+/**
+ * @summary The config to be used. Can be updated via: setConfig
+ */
+let _fledgeConfig = {};
+
+/**
+ *
+ * @param {Object} reqBidsConfigObj required; This is the same param that's used in pbjs.requestBids.
+ * @param {function} fn required; The next function in the chain, used by hook.js
+ */
+export function renderAdHook(fn, doc, id, options) {
+  const bid = auctionManager.findBidByAdId(id);
+
+  if (_fledgeConfig.auctionConfig) {
+    const fledge = new Fledge('https://magniteengineering.github.io/fledge.polyfill/iframe.html');
```

Comment:
Why is external code needed here? Also, please lock https://magniteengineering.github.io/fledge.polyfill/scripts/frame/esm/index.js to a version if external code is actually required. Also, since it requires promises / async await, please make sure the browser can handle these features before adding these requests.

Reply:
> Why is external code needed here?

The external code is a polyfill that is required to work with Google's new Fledge proposal.

> Also, please lock https://magniteengineering.github.io/fledge.polyfill/scripts/frame/esm/index.js to a version if external code is actually required.

This is locked down in the `package.json` to a version (https://github.com/prebid/Prebid.js/pull/7189/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R105)

> Also, since it requires promises / async await, please make sure the browser can handle these features before adding these requests.

Do you have any suggestions for how to handle this?  Do you guys have/use some sort of agent/browser sniffing that I could work with in order to prevent the code from running in older browsers?  Technically this piece of code will only work with Chrome at version 91 or later.

Reply:
I am confused, if you don't need the runtime code bc you included it at build time, why are you invoking it. 

I would suggest not sniffing for user agents or browser versions but just checking for the existence of the required api or functionality

Reply:
> I am confused, if you don't need the runtime code bc you included it at build time, why are you invoking it.

I'm also confused by the concern and questions here.  


I included the `@magnite/fledge.polyfill` library because it's a polyfill that is required to work with Fledge.

> I would suggest not sniffing for user agents or browser versions but just checking for the existence of the required api or functionality

okay, thanks.

Reply:
yeah, this is new stuff that doesn't exist yet

Reply:
That is intentional @ChrisHuie Chrome 91 will be the version of Chrome that supports Fledge. We have to match this "module" with requests from the right Chrome browser version.

Reply:
well Chrome 94 is out now, so it will be much later at this point

Reply:
right I guess the pt is that this module is intended to only work on a very specific version of Chrome, TBD on which version.